### PR TITLE
Add api key settings

### PR DESCRIPTION
**Pull request by nelsonkopliku**
api-key-ok
api-key-ko

### CODE REVIEW

**Review by fabriziosestito — APPROVED** [review]
this looks just great! thanks!

File: assets/js/components/Settings/Settings.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from 'react';
+import { get } from 'axios';
+
+import Button from '@components/Button';
+
+import { logError } from '@lib/log';
+
+import { Transition } from '@headlessui/react';
+import classNames from 'classnames';
```
Comment (dottorblaster) [inline assets/js/components/Settings/Settings.jsx:9]: Can you move these two imports on top?

File: assets/js/components/Settings/Settings.jsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        apiKey !== undefined && setApiKey(apiKey);
+        setLoading(false);
+      })
+      .catch((error) => {
+        logError(error);
+        setLoading(false);
+      });
+  }, []);
+
+  const hasApiKey = apiKey !== null;
```
Comment (dottorblaster) [inline assets/js/components/Settings/Settings.jsx:29]: Would eslint complain if we put `Boolean(apiKey)` here?

**Review by dottorblaster — APPROVED** [review]
Just a couple comments

File: assets/js/components/Settings/Settings.jsx (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/components/Settings/Settings.jsx:9]: Ok. like this?
```
import React, { useState, useEffect } from 'react';
import { Transition } from '@headlessui/react';
import classNames from 'classnames';

import { get } from 'axios';
import Button from '@components/Button';
import { logError } from '@lib/log';
```

File: assets/js/components/Settings/Settings.jsx (hunk 2 above)
Comment (nelsonkopliku) [reply assets/js/components/Settings/Settings.jsx:29]: does not complain it seems. Gonna change in one commit with the previous comment.

**Pull request merged**